firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -912,7 +915,8 @@ Map<String, Object> serialize(T object, ErrorPath path) {
     }
 
     private void applyFieldAnnotations(Field field) {
-      if (field.isAnnotationPresent(ServerTimestamp.class)) {
+      if (field.isAnnotationPresent(ServerTimestamp.class)
+          || isAnnotationPresentOnProperty(field, ServerTimestamp.class)) {
```

Comment:
The trick is to use Kotlin reflection do another round of reflection to get annotations on properties.